code-challenges.js

```diff
@@ -45,16 +83,44 @@ var stringWithVowels2 = "ILoveJavaScript"
 var notAString1 = true
 var notAString2 = 42
 
+const deVowelizer = (string)  => {
 
+    if( typeof string === "string") {
+        return "true"
+    } else if (typeof string !== "string") {
+        return "not a string"
+    }
+    let vowels = ["a", "e", "i", "o", "u", "A", "E", "I", "O", "U"]
+    let newArry = string.split("")
+        
+    return newArry.filter(value => vowels.indexOf(value) === -1).join("")
+}
 
+console.log(deVowelizer(notAString1))
 
 // --------------------6) Create a function that takes the toonimals array and returns an array with only the toon objects that are cats.
 // Expected output: [ { name: "Stimpy", animal: "cat" }, { name: "Scratchy", animal: "cat" }, { name: "Felix", animal: "cat" } ]
 
 
-var toonimals = [ {name: "Itchy", animal: "mouse"}, {name: "Stimpy", animal: "cat"}, {name: "Daffy", animal: "duck"}, {name: "Scratchy", animal: "cat"}, {name: "Ren", animal: "dog"}, {name: "Felix", animal: "cat"}]
+var toonimals = [ {name: "Itchy", animal: "mouse"}, {name: "Stimpy", animal: "cat"}, {name: "Daffy", animal: "duck"}, {name: "Scratchy", animal: "cat"}, {name: "Ren", animal: "dog"}, {name: "Felix", animal: "cat"}] 
 
+// create function that takes in an array as an argument
+const catify = (array) => {
+     return array.filter(value => value.animal === "cat")
+}
 
+console.log(catify(toonimals))
+// use filter to iterate and filter out the non cat objects
 
 // --------------------7) Using the toonimals variable from #6, create a function that returns only the names of the non-cats.
 // Expected output: "Itchy" "Daffy" "Ren"
+
+const catify = (array) => {
+     array.filter(value => {
+         if (value.animal !== "cat") {
+              console.log(value.name);
+         }
+        })
+}
+
```

Comment:
every function requires a return. i see what your train of thought was here but you'll need to sub out that console.log() and find a place for a return. hint: you'll probably want to chain a `.map()` onto your `.filter()`